specification/iotoperations/IoTOperations.Management/models/akri/common.tsp

```diff
@@ -0,0 +1,219 @@
+  /** The name of the image pull secret. */
+  secretRef: string;
+}
+
+/** AkriConnectorsTagDigestSettings properties. */
+@added(Versions.`2025-07-01-preview`)
+@discriminator("tagDigestType")
+model AkriConnectorsTagDigestSettings {
+  /** The tag or digest type. */
+  tagDigestType: AkriConnectorsTagDigestType;
+}
+/** AkriConnectorsTagDigestType values. */
+@added(Versions.`2025-07-01-preview`)
+union AkriConnectorsTagDigestType {
+  string,
+
+  /** Indicates that a tag should be specified. */
+  Tag,
+
+  /** Indicates that a digest should be specified. */
+  Digest,
+}
+
+/** AkriConnectorsTag properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorsTag extends AkriConnectorsTagDigestSettings {
+  /** The tag or digest type. */
+  tagDigestType: Tag;
+
+  /** The tag of the image. */
+  tag: string;
+}
+
+/** AkriConnectorsDigest properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorsDigest extends AkriConnectorsTagDigestSettings {
+  /** The tag or digest type. */
+  tagDigestType: Digest;
+
+  /** The digest of the image. */
+  digest: string;
+}
+/** AkriConnectorsRegistrySettings properties. */
+@added(Versions.`2025-07-01-preview`)
+@discriminator("registrySettingsType")
+model AkriConnectorsRegistrySettings {
+  registrySettingsType: AkriConnectorsRegistrySettingsType;
+}
+
+/** AkriConnectorsRegistrySettings properties. */
+@added(Versions.`2025-07-01-preview`)
+union AkriConnectorsRegistrySettingsType {
+  string,
+
+  /** A Registry Endpoint reference. */
+  RegistryEndpointRef,
+
+  /** A Container Registry reference. */
+  ContainerRegistry,
+}
+
+/** AkriConnectorsRegistryEndpointRef properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorsRegistryEndpointRef extends AkriConnectorsRegistrySettings {
+  /** The registry endpoint reference. */
+  registrySettingsType: RegistryEndpointRef;
+
+  /** The name of the registry endpoint. */
+  registryEndpointRef: string;
+}
+
+/** AkriConnectorsContainerRegistry properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorsContainerRegistry extends AkriConnectorsRegistrySettings {
+  /** The registry settings type. */
+  registrySettingsType: ContainerRegistry;
+
+  /** The registry settings for the container registry. */
+  containerRegistrySettings: AkriConnectorsContainerRegistrySettings;
+}
+
+/** AkriConnectorsContainerRegistry properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorsContainerRegistrySettings {
+  /** The container registry to use for the artifact. */
+  registry: string;
+
+  /** Optional list of references to secrets in the same namespace to use for pulling the connector image. */
+  @OpenAPI.extension("x-ms-identifiers", #["secretRef"])
+  imagePullSecrets?: AkriConnectorsImagePullSecret[];
+}
+
+/** AkriConnectorsSecret properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorsSecret {
+  /** The key in the secret to be mounted. */
+  secretKey: string;
```

Comment:
Wondering: does this need to be annotated as @secret or something? So that its x-ms-secret in the open api spec?